src/layers/fp64/scatterplot-layer/scatterplot-layer.js

```diff
@@ -0,0 +1,195 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+import {BaseLayer} from '../../../lib';
+import {Model, Program, Geometry} from 'luma.gl';
+
+const glslify = require('glslify');
+const DEFAULT_COLOR = [255, 0, 255];
+
+const defaultGetPosition = x => x.position;
+const defaultGetRadius = x => x.radius;
+const defaultGetColor = x => x.color || DEFAULT_COLOR;
+
+export default class ScatterplotLayer extends BaseLayer {
+  /*
+   * @classdesc
+   * ScatterplotLayer
+   *
+   * @class
+   * @param {object} props
+   * @param {number} props.radius - point radius
+   */
+  constructor({
+    getPosition = defaultGetPosition,
+    getRadius = defaultGetRadius,
+    getColor = defaultGetColor,
+    ...props
+  }) {
+    super({
+      getPosition,
+      getRadius,
+      getColor,
+      ...props
+    });
+  }
+
+  initializeState() {
+    const {gl} = this.state;
+    const {attributeManager} = this.state;
+
+    this.setState({
+      model: this.getModel(gl)
+    });
+
+    attributeManager.addInstanced({
+      instancePositions: {size: 4, update: this.calculateInstancePositions},
+      instanceRadius: {size: 1, update: this.calculateInstanceRadius},
+      layerHeight: {size: 2, update: this.calculateLayerHeight},
+      instanceColors: {size: 3, update: this.calculateInstanceColors}
+    });
+  }
+
+  didMount() {
+    this.updateUniforms();
+  }
+
+  willReceiveProps(oldProps, newProps) {
+    super.willReceiveProps(oldProps, newProps);
+    this.updateUniforms();
+  }
+
+  getModel(gl) {
+    const NUM_SEGMENTS = 16;
+    const PI2 = Math.PI * 2;
+
+    let positions = [];
+    for (let i = 0; i < NUM_SEGMENTS; i++) {
+      positions = [
+        ...positions,
+        Math.cos(PI2 * i / NUM_SEGMENTS),
+        Math.sin(PI2 * i / NUM_SEGMENTS),
+        0
+      ];
+    }
+
+    return new Model({
+      program: new Program(gl, {
+        vs: glslify('./scatterplot-layer-vertex.glsl'),
+        fs: glslify('./scatterplot-layer-fragment.glsl'),
+        id: 'fp64-scatterplot'
+      }),
+      geometry: new Geometry({
+        drawMode: 'TRIANGLE_FAN',
+        positions: new Float32Array(positions)
+      }),
+      isInstanced: true
+    });
+  }
+
+  df64ify(a) {
+    const a_hi = new Float32Array([a])[0];
+    const a_lo = a - a_hi;
+    return [a_hi, a_lo];
+  }
+
+  updateUniforms() {
+    this.calculateZoomRadiusFP64();
```

Comment:
Not sure that we need to change the name of this method. The fact that the zoom radius happens to be stored in 64 bits is a detail inside that function.